lib/helpers.js

```diff
@@ -0,0 +1,113 @@
+const _ = require("lodash");
+const parsers = require("./parsers");
+const consts = require("./consts.json");
+
+function removeCredentials(params, labels = consts.DEFAULT_CREDENTIAL_LABELS) {
+  return { ..._.omit(params, labels) };
+}
+
+function loadConfiguration() {
+  try {
+    // eslint-disable-next-line global-require
+    return require("../../../config.json");
+  } catch (exception) {
+    console.error(exception);
+    throw new Error("Could not retrieve the plugin configuration");
+  }
+}
+
+function readRegion(params, settings, label = consts.DEFAULT_CREDENTIAL_LABELS[2]) {
+  return parsers.autocomplete(params[label]) || parsers.autocomplete(settings[label]);
+}
+
+function readCredentials(params, settings, labels = consts.DEFAULT_CREDENTIAL_LABELS) {
+  return {
+    accessKeyId: parsers.string(params[labels[0]]) || parsers.string(settings[labels[0]]),
+    secretAccessKey: parsers.string(params[labels[1]]) || parsers.string(settings[labels[1]]),
+    region: readRegion(params, settings),
+  };
+}
+
+function removeUndefinedAndEmpty(object) {
+  return _.omitBy(object, (v) => _.isNil(v) || (_.isObjectLike(v) && _.isEmpty(v)));
```

Comment:
`value` makes leaves no doubt on what the function parameter is.

Reply:
PS. I have nothing against shortening param types to `v` in cases where it's widely known that the parameter is a value (like. `Object.entries`), but in this case it's our own data structure, not a standard.